### PR TITLE
Add a C# sample for post call analytics

### PR DESCRIPTION
**Pull request by mhko**
## Purpose
This C# sample uses the new Fast Transcription API and Azure OpenAI SDK to summarize an audio recording. Usage is as below. 

```
.\post-call-analytics.exe --speechKey *** --speechRegion eastus --openAiKey *** --openAiEndpoint https://***.openai.azure.com --openAiDeploymentName gpt-4o-mini --inputAudio Call1_separated_16k_health_insurance.wav
https://eastus.api.cognitive.microsoft.com/speechtotext/transcriptions:transcribe?api-version=2024-05-15-preview : OK
Transcription: Hello, thank you for calling Contoso. ...
https://openainateko.openai.azure.com : 200
Summary: Mary Rondo called Contoso to enroll in health insurance. After providing her personal information, including her social security number and email address, she inquired about receiving a physical membership card, which the representative confirmed could be mailed to her address in Lansing.
```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[X] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

**Review by vitastark — APPROVED** [review]

**Review by Emma-ms — DISMISSED** [review]

**Review by gkatuka — APPROVED** [review]

File: scenarios/csharp/dotnetcore/post-call-analytics/Program.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        internal static async Task<string> SummarizeAsync(string openAiKey, string openAiEndpoint, string deploymentOrModelName, string transcription)
+        {
+            var azureClient = new AzureOpenAIClient(new Uri(openAiEndpoint), new ApiKeyCredential(openAiKey));
+            var chatClient = azureClient.GetChatClient(deploymentOrModelName);
+
+            var prompt = $"You are an AI assistant that helps extract information from customer call center transcripts. " +
+                $"Summarize the conversation in a couple sentences. \n\"\"\"\n{transcription}\n\"\"\"\n";
+
+            var completion = await chatClient.CompleteChatAsync(
+                new SystemChatMessage(prompt)
```
Comment (rhurey) [inline scenarios/csharp/dotnetcore/post-call-analytics/Program.cs:66]: It feels like the system message is the constant part of this prompt, but the transcription should be a user message.

File: scenarios/csharp/dotnetcore/post-call-analytics/Program.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            );
+
+            Console.WriteLine($"{openAiEndpoint} : {completion.GetRawResponse().Status}");
+
+            var summary = completion.Value.Content[0].Text;
+            return summary;
+        }
+
+        internal static async Task AnalyzeAudioAsync(string speechKey, string speechRegion, FileInfo? inputAudio, string openAiKey, string openAiEndpoint, string deploymentOrModelName)
```
Comment (rhurey) [inline scenarios/csharp/dotnetcore/post-call-analytics/Program.cs:75]: Why support a nullable FileInfo, and then error if it's null?

**Review by rhurey — APPROVED** [review]
Some minor comments.

File: scenarios/csharp/dotnetcore/post-call-analytics/Program.cs (hunk 2 above)
Comment (mhko) [reply scenarios/csharp/dotnetcore/post-call-analytics/Program.cs]: Removed ? and added a check for inputAudio.Exists

**Review by rhurey — APPROVED** [review]

**Review by rhurey — APPROVED** [review]

**Pull request merged**